gui/src/components/TopBar.tsx

```diff
@@ -194,7 +195,10 @@ export function TopBar({
               </div>
               <div
                 className="flex items-center justify-center hover:bg-background-60 rounded-full w-7 h-7"
-                onClick={() => appWindow.close()}
+                onClick={async () => {
+                  await invoke('update_window_state');
+                  await appWindow.close();
```

Comment:
why do you await the close here? Does `.close()` return when the app window gets exited?

Reply:
really weird nitpick, i dont know

Reply:
my comment was a question not a nitpick :) If it is waiting for close, then this doensn't make much sense because it just creates a lot of promises that will never complete until the application exits. But maybe I'm misunderstanding something, which is why I asked